Objects/codeobject.c

```diff
@@ -443,12 +443,14 @@ _PyCode_Validate(struct _PyCodeConstructor *con)
      * usually prevents crashes due to assertions but a MemoryError may still
      * be triggered later.
      *
-     * See https://github.com/python/cpython/issues/126119 for details.
+     * See https://github.com/python/cpython/issues/126119 for details
+     * and corresponding PR for the rationale on the upper limit value.
      */
-    int ub = (int)(INT_MAX / sizeof(PyObject *)) - FRAME_SPECIALS_SIZE;
+    Py_ssize_t limit = (Py_ssize_t)(INT_MAX / 16) - FRAME_SPECIALS_SIZE;
```

Comment:
No need for the `FRAME_SPECIALS_SIZE`. That's why it is `INT_MAX / 16` not `INT_MAX/8`, to make sure there is *plenty* of headroom.